akka-persistence-typed/src/main/scala/akka/persistence/typed/dstate/internal/RequestingRecoveryPermit.scala

```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.dstate.internal 
+class RequestingRecoveryPermit {
+
+}
```

Comment:
I wonder if this will be needed, the recovery permit is used for event sourced journal, but this will be more like or potentially use the snapshot store api rather, which does not have recovery permits (as it is a single read per "recovery")

Reply:
I think it can overload the database in similar way as for event sourcing. Many "entities" starting at the same time (e.g. shard rebalance) will trigger many reads.

The recovery permit for EventSourcedBehavior isn't for events vs snapshot, it's before the the actor starts any kind of recovery.

Reply:
Ah, my bad, thx for the clarification.